File: apps/ios/OpenFrame/Model/Recipe.swift
import Foundation

struct Recipe: Identifiable, Codable, Sendable, Equatable {
    let id: String
    let title: String
    let description: String?
    let servings: Int?
    let prepTime: Int?
    let cookTime: Int?
    let ingredients: [RecipeIngredient]
    let instructions: [String]
    let tags: [String]
    let notes: String?
    let sourceImagePath: String?
    let thumbnailPath: String?
    let isFavorite: Bool

    var totalTime: Int? {
        if let prep = prepTime, let cook = cookTime { return prep + cook }
        return prepTime ?? cookTime
    }
}

struct RecipeIngredient: Codable, Sendable, Equatable, Identifiable {
    var id: String { "\(name)-\(amount)-\(unit)" }
    let name: String
    let amount: String
    let unit: String

    var displayText: String {
        var parts: [String] = []
        if !amount.isEmpty { parts.append(amount) }
        if !unit.isEmpty { parts.append(unit) }
        parts.append(name)
        return parts.joined(separator: " ")
    }
}

File: apps/android/app/src/main/java/us/openframe/app/domain/model/Recipe.kt
package us.openframe.app.domain.model

data class Recipe(
    val id: String,
    val title: String,
    val description: String? = null,
    val servings: Int? = null,
    val prepTime: Int? = null,
    val cookTime: Int? = null,
    val ingredients: List<RecipeIngredient> = emptyList(),
    val instructions: List<String> = emptyList(),
    val tags: List<String> = emptyList(),
    val notes: String? = null,
    val sourceImagePath: String? = null,
    val thumbnailPath: String? = null,
    val isFavorite: Boolean = false,
) {
    val totalTime: Int? get() = when {
        prepTime != null && cookTime != null -> prepTime + cookTime
        prepTime != null -> prepTime
        cookTime != null -> cookTime
        else -> null
    }
}

data class RecipeIngredient(
    val name: String,
    val amount: String = "",
    val unit: String = "",
) {
    val displayText: String get() = buildString {
        if (amount.isNotBlank()) append("$amount ")
        if (unit.isNotBlank()) append("$unit ")
        append(name)
    }.trim()
}
